cardano-cli/src/Cardano/CLI/EraBased/Run/Genesis/CreateTestnetData.hs

```diff
@@ -338,40 +340,41 @@ runGenesisCreateTestNetDataCmd
 
     addDRepsToConwayGenesis
       :: [VerificationKey DRepKey]
+      -- \^ The credential of the DReps
       -> [VerificationKey StakeKey]
+      -- \^ The credentials of those that delegate their votes to the dreps. It happens
+      -- to only be stake keys now, but it could be more general in the future.
       -> L.ConwayGenesis L.StandardCrypto
+      -- \^ The genesis data to amend
       -> L.ConwayGenesis L.StandardCrypto
     addDRepsToConwayGenesis dRepKeys stakingKeys conwayGenesis =
-      conwayGenesis
-        { L.cgDelegs = delegs (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
-        , L.cgInitialDReps = initialDReps (L.ucppDRepDeposit $ L.cgUpgradePParams conwayGenesis) dRepKeys
-        }
+      conwayGenesis{L.cgDelegs, L.cgInitialDReps}
      where
-      delegs
-        :: [(VerificationKey StakeKey, VerificationKey DRepKey)]
-        -> ListMap (L.Credential L.Staking L.StandardCrypto) (L.Delegatee L.StandardCrypto)
+      -- The credential, to the drep it delegates to
+      delegs :: [(L.Credential L.Staking L.StandardCrypto, VerificationKey DRepKey)]
       delegs =
-        fromList
-          . map
-            ( bimap
-                verificationKeytoStakeCredential
-                (L.DelegVote . L.DRepCredential . verificationKeyToDRepCredential)
-            )
+        map
+          (first verificationKeytoStakeCredential)
+          (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
```

Comment:
This is problematic and thankfully it showed up in the tests:
```
/out/conway-genesis.json
        149 ┃ 
        150 ┃     length (cgInitialDReps conwayGenesis) H.=== numDReps
            ┃     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            ┃     │ ━━━ Failed (- lhs) (+ rhs) ━━━
            ┃     │ - 4
            ┃     │ + 5
        151 ┃ 
        152 ┃     length (cgDelegs conwayGenesis) H.=== numStakeDelegs
```
We have: `numStakeDelegs = 4`  and 
```
numDReps :: Int
numDReps = 5
```

Because we use `zip` the 5th drep is ignored because there is no stake key to delegate to it. We must therefore take into consideration dreps with no delegators.

Reply:
Indeed, one more win of testing :laughing: 

I corrected this :+1: